File: leetcode/1302.层数最深叶子节点的和.py
#
# @lc app=leetcode.cn id=1302 lang=python3
#
# [1302] 层数最深叶子节点的和
#
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
# @lc code=start


class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        stack = [root]

        while stack:
            new_stack = [
                child
                for node in stack
                if node
                for child in (node.left, node.right)
                if child
            ]
            if not new_stack:
                break
            stack = new_stack
        return sum(node.val for node in stack)
# @lc code=end

File: leetcode/1302.层数最深叶子节点的和.cpp
#include <iostream>
#include <queue>
using namespace std;
/*
 * @lc app=leetcode.cn id=1302 lang=cpp
 *
 * [1302] 层数最深叶子节点的和
 */

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

// @lc code=start

class Solution
{
public:
    int deepestLeavesSum(TreeNode *root)
    {
        queue<TreeNode *> qu;
        qu.emplace(root);
        int res;
        while (qu.size() > 0)
        {
            res = 0;
            int count = qu.size();
            for (size_t i = 0; i < count; i++)
            {
                auto node = qu.front();
                qu.pop();
                res += node->val;
                if (node->left != nullptr)
                {
                    qu.emplace(node->left);
                }
                if (node->right != nullptr)
                {
                    qu.emplace(node->right);
                }
            }
        }
        return res;
    }
};
// @lc code=end
